flytekit/types/directory/types.py

```diff
@@ -168,7 +168,7 @@ def __init__(
         """
         # Make this field public, so that the dataclass transformer can set a value for it
         # https://github.com/flyteorg/flytekit/blob/bcc8541bd6227b532f8462563fe8aac902242b21/flytekit/core/type_engine.py#L298
-        self.path = path
+        self.path = path if not isinstance(pathlib.Path, str) else str(path)
```

Comment:
Is `not isinstance(pathlib.Path, str)` intended? It is always True. The type `pathlib.Path` is never a `str`.

Also, doing any conversion here is backward incompatible. If a user passed in a `Path` object before, then doing `str(path)` will change the type.

Do we need `str(path)` for this PR?